modules/apps/27-interchain-accounts/host/keeper/handshake.go

```diff
@@ -156,16 +181,22 @@ func (k Keeper) OnChanUpgradeTry(ctx sdk.Context, portID, channelID string, orde
 		return "", err
 	}
 
-	if err := icatypes.ValidateHostMetadata(ctx, k.channelKeeper, connectionHops, metadata); err != nil {
+	// ValidateHostMetadata will ensure the ICS27 protocol version has not changed and that the
```

Comment:
also checks controller/host connection IDs to those provided, might be worth adding this here if we also add the inline comments for the explicit checks for this in lines 196 and 200.

I guess it followed along with the docu's added for controller in which case we should maybe update both!